src/main/java/uk/gov/hmcts/reform/unspec/service/OrganisationService.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.reform.unspec.service;
+
+import feign.FeignException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.prd.client.OrganisationApi;
+import uk.gov.hmcts.reform.prd.model.Organisation;
+
+import java.util.Optional;
+
+import static java.util.Optional.ofNullable;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class OrganisationService {
+
+    private final OrganisationApi organisationApi;
+    private final AuthTokenGenerator authTokenGenerator;
+
+    public Optional<Organisation> findOrganisation(String authToken) {
+        try {
+            return ofNullable(organisationApi.findUserOrganisation(authToken, authTokenGenerator.generate()));
+
+        } catch (FeignException.NotFound | FeignException.Forbidden ex) {
+            log.error("User not registered in MO", ex);
```

Comment:
Is there some potentially useful data in response body? If yes, then it's better to log `ex.contentUTF8()`, because FeignException does not show it by default.

Reply:
Possible errors from this endpoint are:

400 | An invalid ID was provided
403 | Forbidden Error: Access denied
404 | No Organisation found with the given ID

We know that 400 in theory should never happen from a logged in user, 403 in theory should also never happen when we are whitelisted but the user could maybe not have the right roles to query this endpoint, so this just leaves the 404. I don't think we are swollowing anything useful there so will leave as is.